### PR TITLE
Conditions dash

### PR DESCRIPTION
**Pull request by agpiermarini**
Changes proposed:
- Add number of methods to conditions model for dashboard; model and feature tests
- Add controllers dashboard view
- Add styling for dashboard view

connects to #29, connects to #116, connects to #117, connects to #118

@adam-conway @anubiskhan @vadlusk @agpiermarini

### CODE REVIEW

File: app/models/condition.rb [hunk 1]
```diff
@@ -1,3 +1,112 @@
 class Condition < ApplicationRecord
   validates :date, presence: true
+
+  def self.trips_by_temperature(*range)
```
Comment (agpiermarini) [inline app/models/condition.rb]: Question re: acceptable balance of AR and Ruby: **If ActiveRecord is handling all the groupings and counts in the conditions dash, is it okay to use Ruby to find the min/max/average of those counts?**  

The `max_trips_by`,`min_trips_by` and `avg_trips_by` methods in this model iterate over the `<attribute>_ranges` hashes, sending the values to one of the four ActiveRecord query methods at the top (e.g. `trips_by_temperature` or `trips_by_precipitation`). The query methods return a hash where keys are dates and the values are trip count. 

This AR hash is then passed to the `return_min`, `return_max`, and `return_avg` methods, which return the highest/lowest/avg value in the hash.

Now, the `return_min` and `return_max` methods _could_ be replaced by building out longer queries.  Such as using the following to get max rides by day by visibility:

```
Condition.joins("JOIN trips ON conditions.date = trips.start_date")
  .select("conditions.date, count(trips.id) as trip_count")
  .where(mean_visibility_miles: [10..20])
  .group("conditions.date")
  .order("trip_count DESC")
  .first
  .trip_count
```
However, I like that in it's current construction, we are always dealing with the same ActiveRecord hash and leaning on some explicit Ruby methods to return the count we need. The query above is arguably more convoluted, all for the purpose of returning an integer from a hash. What are your thoughts on how this has been put together?

File: app/models/condition.rb [hunk 2]
```diff
@@ -1,3 +1,112 @@
 class Condition < ApplicationRecord
   validates :date, presence: true
+
```
Comment (agpiermarini) [inline app/models/condition.rb]: Question re: acceptable balance of AR and Ruby: **If ActiveRecord is handling all the groupings and counts in the conditions dash, is it okay to use Ruby to find the min/max/average of those counts?**  

The `max_trips_by`,`min_trips_by` and `avg_trips_by` methods in this model iterate over the `<attribute>_ranges` hashes, sending the values to one of the four ActiveRecord query methods at the top (e.g. `trips_by_temperature` or `trips_by_precipitation`). The query methods return a hash where keys are dates and the values are trip count. 

This AR hash is then passed to the `return_min`, `return_max`, and `return_avg` methods, which return the highest/lowest/avg value in the hash.

Now, the `return_min` and `return_max` methods _could_ be replaced by building out longer queries.  Such as using the following to get max rides by day by visibility:

```
Condition.joins("JOIN trips ON conditions.date = trips.start_date")
  .select("conditions.date, count(trips.id) as trip_count")
  .where(mean_visibility_miles: [10..20])
  .group("conditions.date")
  .order("trip_count DESC")
  .first
  .trip_count
```
However, I like that in it's current construction, we are always dealing with the same ActiveRecord hash and leaning on some explicit Ruby methods to return the count we need. The query above is arguably more convoluted, all for the purpose of returning an integer from a hash. What are your thoughts on how this has been put together?

File: app/models/condition.rb [hunk 3]
```diff
@@ -1,3 +1,112 @@
...
+  end
+
+  def self.avg_trips_by(ranges, trips)
+    breakdown = Hash.new(0)
+    ranges.each do |key, value|
+      breakdown[key] = return_avg(send(trips, value))
+    end
+    breakdown
+  end
+
```
Comment (agpiermarini) [inline app/models/condition.rb]: By way of example, `avg_trips_by` returns a hash that looks like this:

```
{"0 mph to 4 mph"=>41,
 "5 mph to 9 mph"=>40,
 "10 mph to 14 mph"=>25,
 "15 mph to 19 mph"=>23,
  "20 mph to 24 mph"=>20,
  "25 mph to 29 mph"=>6,
  "30 mph to 34 mph"=>5}
```

**Comment by anubiskhan** [comment]
Great stuff, I think going with Ruby in these methods is a solid and acceptable choice.

**Comment by anubiskhan** [comment]
Anyway to have AR handle the average trips per day en lieu of Ruby? We can save it for refactor though

**Pull request merged**